tracer/src/Datadog.Trace/Iast/ReturnedHeadersAnalyzer.cs

```diff
@@ -38,25 +41,60 @@ internal static void Analyze(IHeaderDictionary responseHeaders, IntegrationId in
         }
 
         IastModule.OnExecutedSinkTelemetry(IastInstrumentedSinks.XContentTypeHeaderMissing);
+        IastModule.OnExecutedSinkTelemetry(IastInstrumentedSinks.HstsHeaderMissing);
         string contentTypeValue = responseHeaders[ContentType];
         string contentOptionValue = responseHeaders[XContentTypeOptions];
+        string strictTransportSecurityValue = responseHeaders[StrictTransportSecurity];
+        string xForwardedProtoValue = responseHeaders[XForwardedProto];
 
-        if (!IsHtmlResponse(contentTypeValue) || IsNoSniffContentOptions(contentOptionValue))
+        if (!IsHtmlResponse(contentTypeValue))
         {
             return;
         }
 
         LaunchXContentTypeOptionsVulnerability(integrationId, serviceName, contentTypeValue, contentOptionValue);
+        LaunchStrictTransportSecurity(integrationId, serviceName, strictTransportSecurityValue, xForwardedProtoValue, protocol);
     }
 
     private static void LaunchXContentTypeOptionsVulnerability(IntegrationId integrationId, string serviceName, string contentTypeValue, string contentOptionValue)
     {
-        if (!string.IsNullOrEmpty(contentTypeValue))
+        if (!string.IsNullOrEmpty(contentTypeValue) && !IsNoSniffContentOptions(contentOptionValue))
         {
             IastModule.OnXContentTypeOptionsHeaderMissing(integrationId, contentOptionValue, serviceName);
         }
     }
 
+    private static void LaunchStrictTransportSecurity(IntegrationId integrationId, string serviceName, string strictTransportSecurityValue, string xForwardedProtoValue, string protocol)
+    {
+        if (!string.Equals(protocol, "https", StringComparison.OrdinalIgnoreCase)
+            && !string.Equals(xForwardedProtoValue, "https", StringComparison.OrdinalIgnoreCase))
+        {
+            return;
+        }
+
+        if (!IsValidStrictTransportSecurityValue(strictTransportSecurityValue))
+        {
+            IastModule.OnStrictTransportSecurityHeaderMissing(integrationId, serviceName);
+        }
+    }
+
+    // Strict-Transport-Security has a valid value when it starts with max-age followed by a positive number (>0),
+    // it can finish there or continue with a semicolon ; and more content.
+    private static bool IsValidStrictTransportSecurityValue(string strictTransportSecurityValue)
+    {
+        if (string.IsNullOrEmpty(strictTransportSecurityValue) || !strictTransportSecurityValue.StartsWith(MaxAgeConst, StringComparison.OrdinalIgnoreCase))
+        {
+            return false;
+        }
+
+        var index = strictTransportSecurityValue.IndexOf(';');
+
+        var maxAge = (index >= 0 ? strictTransportSecurityValue.Substring(MaxAgeConst.Length, index - MaxAgeConst.Length) :
+            strictTransportSecurityValue.Substring(MaxAgeConst.Length));
+
+        return (ulong.TryParse(maxAge, out var maxAgeInt));
```

Comment:
If I'm not mistaken, the current logic will return `true` when `maxAgeInt==0`. I think we need the following suggestion, can you confirm?

```suggestion
        return ulong.TryParse(maxAge, out var maxAgeInt) && maxAgeInt > 0;
```

Reply:
A value of 0 is correct but does not provide a practical benefit. Small values, also, like 1, could suffer similar issues. Values like that would show that the header is not missing and, for some reason, it has been decided to provide those values but the header was taken into account. I don't think that a value of 0 should launch a missing header vulnerability, but still, I will bring the topic to the next IAST meeting that we will hold in a few minutes.

Reply:
After discussion, we will implement the same solution that the java team has implemented: to launch a vulnerability if a 0 value is present. There is a pending discussion to be a little bit more accurate when deciding when a value is considered insecure and that solution will be implemented for all the libraries in the future, but, for now, this is the agreed approach. Thanks for bringing this topic up.